lib/ui/prompt.js

```diff
@@ -76,9 +76,11 @@ PromptUI.prototype.processQuestion = function (question) {
 };
 
 PromptUI.prototype.fetchAnswer = function (question) {
+  var screen = this.createScreen();
   var Prompt = this.prompts[question.type];
-  var prompt = new Prompt(question, this.rl, this.answers);
+  var prompt = new Prompt(question, screen, this.rl, this.answers);
```

Comment:
By Prompt, do you mean PromptUI or Prompt? Either way, this would only make sense if we go all the way and have the screen manager do all the cleanup (unmuting and ending output etc). Because now, the order of things is quite specific. And if we were to replace the `releaseCursor()` call below with a `cleanup()` call, it's not exactly clear what the responsibilities of that `cleanup` method are.

```js
UI.prototype.close = function() {
  // ..
  this.rl.output.unmute();

  if (this.currentScreen) {
    this.currentScreen.releaseCursor();
  }

  this.rl.output.end();
  this.rl.pause();
  this.rl.close();
};
```